jobsdb/jobsdb.go

```diff
@@ -84,6 +84,7 @@ type GetQueryParamsT struct {
 	StateFilters                  []string
 	JobCount                      int
 	EventCount                    int
+	PayloadSize                   *int64
```

Comment:
why `*int64` instead of `int`?

Reply:
In case of single-tenant server, the same params is passed to `GetToRetry`, `GetWaiting` and `GetUnprocessed`. I didn't want to have to go over the jobs again and update the payloadLimit, instead I chose to just update it while scanning the rows into jobs. A pointer made that convenient.
Ideally a pointer to the entire params can be passed, which would solve majority of such problems.( looking at having to do `toQuery -= len(retryList)` between `GetToRetry` and `GetWaiting` and `GetUnprocessed`

Reply:
Then this is something that needs to be made clear in the doc, at least.
My personal opinion is that it is unintuitive and maybe an anti-pattern to update a requested query parameter while scanning the results. If we need to communicate this information back to the caller, we could do this by including it in the return value(s)

Reply:
correction: it's not being updated while scanning, but after the scan is complete(refer line 3663 below).
Also, please elaborate why it's an anti-pattern.

Reply:
Function parameters are usually best to be considered as immutable and a function communicates information to the caller using its output(s). Pure functions is something that we should aim for & pure functions don't mutate state or parameters.
A function mutating variables outside of its scope can cause unexpected results, thus is considered to be dangerous. Definitely, any function mutating its parameters should at least explicitly document this extensively.

As an everyday example, imagine that you go to a pizza shop and you write down your (peculiar) order: "I want up to 3 pizzas, but they shouldn't be any more than 1000kcal all together". What you expect from the shop is to get the receipt, your pizzas and maybe the number of their calories. You don't expect that they will get your order, strikethrough the number 1000kcal and replaced it with the remaining calories :) Wouldn't that be awkward?